core/src/main/java/com/scalar/db/api/CrudOperable.java

```diff
@@ -131,4 +155,31 @@ public interface CrudOperable<E extends TransactionException> {
    * @throws E if the transaction CRUD operation fails
    */
   void mutate(List<? extends Mutation> mutations) throws E;
+
+  /** A scanner abstraction for iterating results. */
+  interface Scanner<E extends TransactionException> extends AutoCloseable, Iterable<Result> {
+    /**
+     * Returns the first result in the results.
+     *
+     * @return the first result in the results
+     * @throws E if the operation fails
+     */
```

Comment:
The Javadoc was copied from `com.scalar.db.api.Scanner`, but I think it could be improved to be clear `one()` iterates over the results.

```suggestion
  /**
   * Returns the next result.
   *
   * @return an Optional containing the next result if available, or empty if no more results
   * @throws E if the operation fails
   */
```